scripts/components/profiles/scatterplot.component.js

```diff
@@ -0,0 +1,181 @@
+      .append('g')
+      .attr('transform', 'translate(' + margin.left + ',' + margin.top + ')');
+
+    this.svg.append('g')
+      .attr('class', 'axis axis--x')
+      .attr('transform', 'translate(0,' + this.height + ')')
+      .call(xAxis);
+
+    this.svg.append('g')
+      .attr('class', 'axis axis-line')
+      .attr('transform', 'translate(0,' + this.height + ')')
+      .call(d3_axis_bottom(this.x).ticks(0).tickSizeOuter(0));
+
+    this.svg.append('g')
+      .attr('class', 'axis axis--y')
+      .call(yAxis);
+
+    this.svg.append('g')
+      .attr('class', 'axis axis-line')
+      .call(d3_axis_left(this.y).ticks(0).tickSizeOuter(0));
+
+    this.circles = this.svg.selectAll('circle')
+      .data(this._getFormatedData(0))
+      .enter()
+      .append('circle')
+      .attr('class', function(d) { return d.nodeId.toString() === this.nodeId ? 'dot current' : 'dot' }.bind(this))
+      .attr('r', 5)
+      .attr('cx', function(d) { return this.x(d.x); }.bind(this))
+      .attr('cy', function(d) { return this.y(d.y); }.bind(this));
+
+    if (this.showTooltipCallback !== undefined) {
+      this.circles.on('mousemove', function(d) {
+        const selectedSwitcher = document.querySelector('.js-scatterplot-switcher.selected span');
+
+        this.showTooltipCallback(
+          d,
+          {
+            name: selectedSwitcher.getAttribute('data-name'),
+            unit: selectedSwitcher.getAttribute('data-unit'),
+          },
+          d3_event.clientX + 10,
+          d3_event.clientY + window.scrollY + 10
+        );
+      }.bind(this))
+      .on('mouseout', function() {
+        this.hideTooltipCallback();
+      }.bind(this));
+    }
+  }
+
+  _renderXswitcher() {
+    const template = stringToHTML(ScatterplotSwitcherTemplate({data: this.xDimension}));
+    this.el.appendChild(template[0]);
+
+    this.switchers = Array.prototype.slice.call(this.el.querySelectorAll('.js-scatterplot-switcher'), 0);
+    this.switchers.forEach(switcher => {
+      switcher.addEventListener('click', (e) => this._switchTab(e));
+    });
+  }
+
+  _switchTab(e) {
+    const selectedSwitch = e && e.currentTarget;
+    if (!selectedSwitch) {
+      return;
+    }
+
+    const selectedTabKey = selectedSwitch.getAttribute('data-key');
+    this.switchers.forEach(switcher => {
+      switcher.classList.remove('selected');
+    });
+    selectedSwitch.classList.add('selected');
+
+    const newData = this._getFormatedData(selectedTabKey);
+    let allXValues = newData.map(item => item.x);
+    let x = d3_scale_linear()
+      .range([0, this.width])
+      .domain(d3_extent([0, ...allXValues]));
+    const xAxis = d3_axis_bottom(x)
+      .ticks(8)
+      .tickSize(-this.height, 0)
+      .tickPadding(9)
+      .tickFormat((value, i) => {
+        if (i === 0) {
+          return null;
+        }
+
+        return abbreviateNumber(value, 3);
+      });
+
+    this.circles
+      .data(newData)
+      .transition()
```

Comment:
So in the end mapping between nodes is made automatically when they have the same nodeId right?